File: Backend/src/Fibertest30.Application/Common/Interfaces/Device/Otau/Otau/OtauDiscover.cs
namespace Fibertest30.Application;

public class OtauDiscover
{
    public string SerialNumber { get;init; } = string.Empty;
    public int PortCount { get; init; }
}

public enum OtauDiscoverError
{
    NoError,
    OsmModuleNotFound,
    UnsupportedOsmModuleConnected,
}

public class OtauDiscoverResult
{
    public OtauDiscover? Discover { get; set; }
    public OtauDiscoverError Error { get; set; } = OtauDiscoverError.NoError;
}

File: WebFt30/src/app/core/store/models/otau-discover.ts
export class OtauDiscover {
  serialNumber!: string;
  portCount!: number;
}

export enum OtauDiscoverError {
  NoError,
  OsmModuleNotFound,
  UnsupportedOsmModuleConnected
}

export class OtauDiscoverResult {
  discover: OtauDiscover | null = null;
  error!: OtauDiscoverError;
}
